DOCS-3472: Add documentation for Viam Apps (#4335)
viamrobotics/docs: docs/operate/get-started/other-hardware/_index.md

</tr>
<tr>
<td><code>models</code></td>
<td>array</td>

> <td>Optional</td>

Reviewer: ```suggestion
<td><strong>Required</strong> if your module provides models</td>
```
[opt] Maybe better to emphasize like this since it's not just plain optional, and this is within the integrate hardware section?

Reviewer 2: It makes that column too long. If anyone makes a mistake with this the CLI will tell them about it and mostly this is auto-generated, so I'm not worried that there will be confusion.

<td>Optional</td>
<td>Enables VS Code hover and autocomplete as you edit your module code. Gets auto-generated when you run <code>viam module generate</code> or <code>viam module create</code>. Has no impact on the module's function.</td>
</tr>
<tr>
<td><code>applications</code></td>
<td>array</td>
<td>Optional</td>

> <td>Objects that provide information about the [apps](/operate/control/viam-app/) associated with the module.</td>

Reviewer: ```suggestion
<td>Objects that provide information about the [apps](/operate/control/viam-app/) provided by the module.</td>
```
